libres/tests/analysis/test_enkf_linalg.cpp

```diff
@@ -2,74 +2,32 @@
 
 #include "ert/analysis/enkf_linalg.hpp"
 
-/**
- * Results generated by this equivalent implementation in numpy:
- *
- * import numpy as np
- * nrow = 2
- * ncol = 3
- *
- * S = np.ones((nrow, ncol))
- * S[0, 1] = 1.5
- * S[1, 0] = 2.5
- *
- * W = np.ones((nrow, ncol))
- * W[0, 1] = 1.3
- *
- * eig = np.array([2.0, 2.0, 3.0])
- *
- * X2 = (W.T @ S)
- * X2 = (np.sqrt(eig) * X2.T).T
- * X2
-*/
-matrix_type *test_enkf_linalg_genX2() {
-    const int nrow = 2;
-    const int ncol = 3;
-
-    matrix_type *X2 = matrix_alloc(ncol, ncol);
-
-    matrix_type *S = matrix_alloc(nrow, ncol);
-    matrix_iset(S, 0, 0, 1.0);
-    matrix_iset(S, 1, 0, 2.5);
-    matrix_iset(S, 0, 1, 1.5);
-    matrix_iset(S, 1, 1, 1.0);
-    matrix_iset(S, 0, 2, 1.0);
-    matrix_iset(S, 1, 2, 1.0);
-
-    matrix_type *W = matrix_alloc(nrow, ncol);
-    matrix_iset(W, 0, 0, 1.0);
-    matrix_iset(W, 1, 0, 1.0);
-    matrix_iset(W, 0, 1, 1.3);
-    matrix_iset(W, 1, 1, 1.0);
-    matrix_iset(W, 0, 2, 1.0);
-    matrix_iset(W, 1, 2, 1.0);
+/*
+Python equivalent:
 
-    const double eig[ncol] = {2.0, 2.0, 3.0};
+W = np.array([[4, 3], [8, 6]])
+D = np.array([[2, 3, 1], [4, 5, 2]])
+eig = np.array([0.5, 0.3, 0.5])
+ens_size = 2
+Lambda = np.diag(eig[0:ens_size])
 
-    enkf_linalg_genX2(X2, S, W, eig);
+X3 = W @ Lambda @ W.T @ D
+*/
+Eigen::MatrixXd test_enkf_linalg_genX3() {
+    // W has dimension min(ens_size, nrobs)
+    Eigen::MatrixXd W{{4.0, 3.0}, {8.0, 6.0}};
```

Comment:
The eigen `comma initializer` allows an alternative syntax for this:
```
Eigen::MatrixXd D(2,2);
D << 4.0, 8.0,
         3.0, 6.0;
```
which "looks more natural" than rows of column based literals in the `{{..}}` initialization.

Reply:
I actually find the comma initialiser less "natural", but this is probably because numpy notation is:

`np.array([[1, 2], [3, 4]])`

and because Mathematica's notation (the best language of course) is

`{{1, 2}, {3, 4}}`